### PR TITLE
Add networking interface rename command

### PR DESCRIPTION
**Pull request by bangelic**
A step is created to show how network interfaces are renamed and
running the installer once more with the new interface names. The
change was required because upgrading using Leapp is inhibited by an
unsupported network configuration.

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.4/Katello 4.6
* [x] Foreman 3.3/Katello 4.5
* [x] Foreman 3.2/Katello 4.4
* [x] Foreman 3.1/Katello 4.3
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.3 or older.

### CODE REVIEW

File: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc [hunk 1]
```diff
@@ -139,6 +139,23 @@ gpgcheck=1
 ----
 endif::[]
 
+ifdef::satellite[]
+. If there are multiple network interfaces, fix the interface names and rerun the installer with the new interface name.
+Call the installer to find out what interface was used and where:
++
+----
+# satellite-installer --help |grep 'interface.*eth'
```
Comment (maximiliankolb) [inline guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc:147]: Please use attributes instead. You'll also need to allow attributes in the code block.

File: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc [hunk 2]
```diff
@@ -139,6 +139,23 @@ gpgcheck=1
...
+    --foreman-proxy-dhcp-interface  DHCP listen interface (current: "eth0")
+    --foreman-proxy-dns-interface  DNS interface (current: "eth0")
+----
+Because `eth0` was renamed to `em0`, call the installer to rename the interface with:
++
+----
+# satellite-installer --foreman-proxy-dhcp-interface=em0 --foreman-proxy-dns-interface=em0
+----
++
```
Comment (maximiliankolb) [inline guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc:156]:
```suggestion
```

**Review by maximiliankolb — COMMENTED** [review]
Can't really comment on the content.

File: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc [hunk 3]
```diff
@@ -139,6 +139,23 @@ gpgcheck=1
 ----
 endif::[]
 
+ifdef::satellite[]
```
Comment (evgeni) [inline guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc:142]: Technically, this is not limited to Satellite and can happen on any deployment.

I would however move this down to the `leapp preupgrade` step, and start off with something like:
If `leapp preupgrade` inhibits the upgrade with "Unsupported network configuration" because there are multiple legacy-named network interfaces, please follow the instructions shown by Leapp to rename the interfaces, followed by an installer run to reconfigure {Project} to use the new interface names.

File: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc [hunk 4]
```diff
@@ -139,6 +139,23 @@ gpgcheck=1
...
+. If there are multiple network interfaces, fix the interface names and rerun the installer with the new interface name.
+Call the installer to find out what interface was used and where:
++
+----
+# satellite-installer --help |grep 'interface.*eth'
+    --foreman-proxy-dhcp-interface  DHCP listen interface (current: "eth0")
+    --foreman-proxy-dns-interface  DNS interface (current: "eth0")
+----
+Because `eth0` was renamed to `em0`, call the installer to rename the interface with:
```
Comment (evgeni) [inline guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc:151]:
```suggestion
If `eth0` was renamed to `em0`, call the installer to use the new interface name with:
```

**Review by evgeni — CHANGES_REQUESTED** [review]

File: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc (hunk 3 above)
Comment (evgeni) [reply guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc:142]: For Satellite builds, we might want to link directly to https://access.redhat.com/solutions/4067471, instead of saying "follow the instructions shown by Leapp" ;)

File: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc [hunk 5]
```diff
@@ -144,12 +144,27 @@ endif::[]
 ----
 # leapp preupgrade
 ----
-The first run is expected to fail but report issues.
+If `leapp preupgrade` inhibits the upgrade with an unsupported network configuration because there are multiple legacy named network interfaces, follow the instructions shown by Leapp to rename the interfaces, followed by an installer run to reconfigure {Project} to use the new interface names:
```
Comment (evgeni) [inline guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc:147]: "Unsupported network configuration" is the output Leapp gives you -- is there a way to highlight it here as such?
In my original suggestion I quoted it, but not sure this is the correct way to do it?

File: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc [hunk 6]
```diff
@@ -144,12 +144,27 @@ endif::[]
 ----
 # leapp preupgrade
 ----
-The first run is expected to fail but report issues.
```
Comment (evgeni) [inline guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc:147]: I think I would keep that here and reword it to something like:
"The first run is expected to report issues and inhibit the upgrade."
And then we can go on and list the possible issues.

Technically, I would probably also merge the next step ("Examine the report") into this one, but probably not in this PR.

File: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc (hunk 5 above)
Comment (bangelic) [reply guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc:147]: @evgeni I can make it bold.

File: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc [hunk 7]
```diff
@@ -144,12 +144,31 @@ endif::[]
...
+----
+If `eth0` was renamed to `em0`, call the installer to use the new interface name with:
++
+[options="nowrap" subs="attributes"]
+----
+# {foreman-installer} --foreman-proxy-dhcp-interface=em0 --foreman-proxy-dns-interface=em0
+----
++
+Continue to the next step for remediation if the first run fails.
++
```
Comment (maximiliankolb) [inline guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc:166]:
```suggestion
```

File: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc [hunk 8]
```diff
@@ -144,12 +144,31 @@ endif::[]
...
 
 . Examine the report in the `/var/log/leapp/leapp-report.txt` file, answer all questions (using `leapp answer`) and manually resolve the other reported problems.
 The following commands show the most common steps required:
 +
+
```
Comment (maximiliankolb) [inline guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc:171]:
```suggestion
```

File: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc [hunk 9]
```diff
@@ -144,12 +144,31 @@ endif::[]
...
+If `leapp preupgrade` inhibits the upgrade with *Unsupported network configuration* because there are multiple legacy named network interfaces, follow the instructions shown by Leapp to rename the interfaces, followed by an installer run to reconfigure {Project} to use the new interface names:
++
+[options="nowrap" subs="attributes"]
+----
+# {foreman-installer} --help |grep 'interface.*eth'
+    --foreman-proxy-dhcp-interface  DHCP listen interface (current: "eth0")
+    --foreman-proxy-dns-interface  DNS interface (current: "eth0")
+----
+If `eth0` was renamed to `em0`, call the installer to use the new interface name with:
```
Comment (maximiliankolb) [inline guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc:158]:
```suggestion

If `eth0` was renamed to `em0`, call the installer to use the new interface name with:
```

**Review by maximiliankolb — COMMENTED** [review]
I am a bit unsure if it's better to go with a nested list or keep it flat.

**Comment by evgeni** [comment]
I'm happy with the current rendering and text!

**Comment by maximiliankolb** [comment]
> I'm happy with the current rendering and text!

OK, then please feel free to merge!

**Review by evgeni — APPROVED** [review]

**Comment by evgeni** [comment]
6ece4a9f8..17c75148f  3.1 -> 3.1
   e36a03040..30d5d1b71  3.2 -> 3.2
   3a9b602a7..74555179d  3.3 -> 3.3
   2faa9a180..4771e467f  3.4 -> 3.4
